js_modules/dagster-ui/packages/ui-core/src/asset-selection/AntlrAssetSelection.ts

```diff
@@ -1,17 +1,46 @@
-import {CharStreams, CommonTokenStream} from 'antlr4ts';
+import {
+  ANTLRErrorListener,
+  CharStreams,
+  CommonTokenStream,
+  RecognitionException,
+  Recognizer,
+} from 'antlr4ts';
 
 import {AntlrAssetSelectionVisitor} from './AntlrAssetSelectionVisitor';
 import {AssetGraphQueryItem} from '../asset-graph/useAssetGraphData';
 import {AssetSelectionLexer} from './generated/AssetSelectionLexer';
 import {AssetSelectionParser} from './generated/AssetSelectionParser';
 
+class AntlrInputErrorListener implements ANTLRErrorListener<any> {
+  syntaxError(
+    recognizer: Recognizer<any, any>,
+    offendingSymbol: any,
+    line: number,
+    charPositionInLine: number,
+    msg: string,
+    e: RecognitionException | undefined,
+  ): void {
+    if (offendingSymbol) {
+      throw new Error(`Syntax error caused by "${offendingSymbol.text}": ${msg}`);
+    }
+    throw new Error(`Syntax error at char ${charPositionInLine}: ${msg}`);
+  }
```

Comment:
I don't think this is necessary tbh, The selection syntax would be the main piece responsible for displaying errors to users:

 https://github.com/dagster-io/dagster/blob/764ea446f503eb8d3c5335d6f351e947f3126d0f/js_modules/dagster-ui/packages/ui-core/src/asset-selection/input/AssetSelectionLinter.ts#L8-L30.

If we want to make a more user friendly error message we should change it there.